src/backend/distributed/commands/database.c

```diff
@@ -293,7 +293,14 @@ PreprocessAlterDatabaseStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	EnsureCoordinator();
+	/* Since ALTER TABLE SET TABLESPACE statement is not supported */
+	/* inside a transaction block, we need to send the command to the */
+	/* main database directly to make it work */
+	if (!IsSetTablespaceStatement(stmt))
+	{
+		EnsureCoordinator();
+	}
+
```

Comment:
Why exactly do we need to throw an error in that case?

Reply:
If not set, I'm getting below error. 
```sql
 NOTICE:  issuing ALTER DATABASE "altered_database!'2" SET TABLESPACE "ts-needs\!escape2"
 DETAIL:  on server postgres@localhost:57638 connectionId: 1
+ERROR:  operation is not allowed on this node
+HINT:  Connect to the coordinator and run it again.
+CONTEXT:  while executing command on localhost:57638
```

When sending Alter database statement other than set tablespace below sql is being executed
```sql
alter database "altered_database!'2" set default_transaction_read_only = true;
NOTICE:  issuing SELECT citus_internal.execute_command_on_remote_nodes_as_user('ALTER DATABASE "altered_database!''2" SET default_transaction_read_only = ''true''', 'postgres')
DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
```